### DISCUSSION TITLE
How to serialize private fields ?

### DISCUSSION THREAD

**Discussion by nthonymiller** [ravendb/ravendb#17602] [General]
How can I serialize a readonly collection of objects?

I've tried doing using a private field with JsonProperty attribute but this doesn't work.

```
public class MyObject 
{
  [JsonProperty("MyLines")]
  private readonly List<IQuoteLine> _lines = new();
  
  [JsonIgnore] 
  public IReadOnlyCollection<IQuoteLine> Lines => _lines.AsReadOnly();

}
```

I'll be adding and removing items via methods.

**Comment by ayende** [comment]
You can use `[ForceJsonSerialization]` on that.

**Reply by nthonymiller** [reply]
I did see the usage of the code in DefaultRavenContractResolver.MembersToFilterOut method, but how can I access this attribute given that it's internal?

**Reply by ppekrol** [reply]
I'm wondering if it would not be best to create a custom converter in DocumentStore.Conventions.Serialization you have two methods:
- CustomizeJsonSerializer
- CustomizeJsonDeserializer

You can add that converter there.

**Reply by murshex** [reply]
ForceJsonSerialization is internal
